physics/module_mp_thompson.F90

```diff
@@ -6471,6 +6488,86 @@ SUBROUTINE semi_lagrange_sedim(km,dzl,wwl,rql,precip,pfsan,dt,R1)
 
   END SUBROUTINE semi_lagrange_sedim
 
+!>\ingroup aathompson
+!! @brief Calculates graupel size distribution parameters
+!!
+!! Calculates graupel intercept and slope parameters for
+!! for a vertical column 
+!!  
+!! @param[in]    kts     integer start index for vertical column
+!! @param[in]    kte     integer end index for vertical column
+!! @param[in]    rand1   real random number for stochastic physics
+!! @param[in]    rg      real array, size(kts:kte) for graupel mass concentration [kg m^3]
+!! @param[out]   ilamg   double array, size(kts:kte) for inverse graupel slope parameter [m]
+!! @param[out]   N0_g    double array, size(kts:kte) for graupel intercept paramter [m-4]
+subroutine graupel_psd_parameters(kts, kte, rand1, rg, ilamg, N0_g)
+
+   implicit none
+
+   integer, intent(in) :: kts, kte
+   real, intent(in) :: rand1
+   real, intent(in) :: rg(:)
+   double precision, intent(out) :: ilamg(:), N0_g(:)
+
+   integer :: k
+   real :: ygra1, zans1
+   double precision :: N0_exp, lam_exp, lamg
+
+   do k = kte, kts, -1
+      ygra1 = alog10(max(1.e-9, rg(k)))
+      zans1 = 3.4 + 2./7.*(ygra1+8.) + rand1
+      N0_exp = 10.**(zans1)
+      N0_exp = max1(dble(gonv_min), min(N0_exp, dble(gonv_max)))
```

Comment:
@grantfirl

Reply:
@AndersJensen-NOAA We settled on changing this to "max" or "dmax1", right? "max1" should be returning an integer that is then cast to double precision, which I don't think is what you want, right?

Reply:
Nevermind, I see that you already did this and this is outdated.